internal/pkg/service/stream/source/type/httpsource/httpsource.go

```diff
@@ -182,15 +182,15 @@ func Start(ctx context.Context, d dependencies, cfg Config) error {
 		// Create connection
 		conn, err := net.Listen("tcp4", cfg.Listen)
 		if err != nil {
-			shutdown(context.Background(), err) // nolint: contextcheck // intentionally creating new context for the shutdown operation
+			shutdown(context.WithoutCancel(ctx), err) // nolint: contextcheck // intentionally creating new context for the shutdown operation
```

Comment:
You can get rid of nolint or ?
```suggestion
			shutdown(context.WithoutCancel(ctx), err)
```